### ISSUE TITLE
Add default for `heron.package.core.uri` to `.heron/conf/aurora/scheduler.yaml` in release

### ISSUE THREAD

**Issue by lilalinda** [twitter/heron#965]
We are currently attempting to set up a Heron multi-node cluster "by hand" (not using Docker/Vagrant) similar to https://github.com/twitter/heron/issues/852 (I believe) and noticed that the default configuration file (probably from heron/config/src/yaml/conf/aurora/scheduler.yaml) is missing an entry for `heron.package.core.uri`.  The documentation does list this setting, though, and when we looked at the logging output, the default location was something like `/vagrant/...` which did not make sense in our environment.

We are currently using
`heron.package.core.uri:                      /home/hadoopuser/.heron/dist/heron-core.tar.gz`
but still having some Aurora issues so your mileage may vary...

**Comment by kramasamy** [comment]
@lilalinda - that configuration should be in client.yaml - not in scheduler.yaml. The documentation needs to be changed. If you could provide a patch that will be great!

**Comment by lilalinda** [comment]
@kramasamy Sorry for my silence -- I was sidetracked with other stuff.  Now back to Heron without vagrant and docker; quite an adventure!  I do have a small cluster of one master and two slave nodes all running Ubuntu 14.04.

I tried your suggestion moving the setting above from scheduler.yaml to client.yaml and also adjusted to URI to include the scheme `file` such that it now says:
`heron.package.core.uri:                      "file:///home/hadoopuser/.heron/dist/heron-core.tar.gz"`
However, I now do get an error on the slave nodes that they cannot find the file, which is not surprising, as I had not installed Heron on them because I naively assumed that the necessary files would be shared through the Hadoop file system.  

Now my specific question: **How should this be set in a non-vagrant case and what is the minimal installation of Heron on the slaves?**  I am reading the `vagrant/init.sh` file with the argument `slave` and it looks like `build_heron()` only gets executed on the master node, which corroborates my assumption above.  I must admit that I am not very familiar with docker yet, so when I look at `install_docker()` I am just reading that it deals with installing `mesos-slave` and starting the service...  Any thoughts or help?

**Comment by kramasamy** [comment]
@lilalinda - When you use ``` file:///``` semantics, it refers to local file. The slave node will not be able to pick up the heron-core.tar.gz - from whichever machine it resides. The key is to place ```heron-core.tar.gz``` in a place where it is accessible from even slave machines. For example, you could place it in a webserver and instead of ```file:///``` you could use ```http://```. Alternatively, you could place it in HDFS and adjust your config ```heron.aurora``` file.

There is a detailed blog post on how to install and run it in Aurora

http://streamanalytics.blogspot.com/2016/06/deploying-heron-on-cluster-of-machines.html
https://github.com/twitter/heron/pull/992

We will continue to refine the documentation so that it is easy to install and run.

**Comment by lilalinda** [comment]
@kramasamy Thanks for the pointer to Supun's blog post -- he did pretty much exactly what I tried (including using HDFS for distributing the Heron core and topology files to the workers).  Likewise, I discovered the same couple of missing libraries (libunwind, libcurl3/4). Unfortunately, I am still running into a runtime issue, which I will post about in a separate thread.  Once I have, I will edit this comment to refer to it.
